kitsune/wiki/utils.py

```diff
@@ -198,3 +200,36 @@ def get_visible_document_or_404(
 
 def get_visible_revision_or_404(user, **kwargs):
     return get_object_or_404(Revision.objects.visible(user, **kwargs))
+
+
+def build_topics_data(request, product, topics):
+    """Build topics_data for use in topic cards"""
+    topics_data = []
+    for topic in topics:
+        docs, _ = documents_for(
+            request.user, request.LANGUAGE_CODE, topics=[topic], products=[product]
+        )
+        docs = [Document.objects.get(id=doc_dict["id"]) for doc_dict in docs]
```

Comment:
```python
docs = Document.objects.filter(id__in=[doc.id for doc in docs]
```

Reply:
Ah - this is good. Better database performance. Thanks for this.